tests/modules/basics.c

```diff
@@ -172,6 +184,9 @@ int TestCallResp3Null(RedisModuleCtx *ctx, RedisModuleString **argv, int argc) {
     reply = RedisModule_Call(ctx,"DEBUG","3cc" ,"PROTOCOL", "null"); /* 3 stands for resp 3 reply */
     if (RedisModule_CallReplyType(reply) != REDISMODULE_REPLY_NULL) goto fail;
 
+    /* make sure we can not reply to resp2 client with resp3 null */
+    if (RedisModule_ReplyWithCallReply(ctx, reply) != REDISMODULE_ERR) goto fail;
+
     RedisModule_ReplyWithSimpleString(ctx,"OK");
```

Comment:
you now have two replies in these functions. that's bad...

Reply:
But the first one verifies that it returns an error and sends nothing... if it succeed the test will fail